manager/integration/tests/common.py

```diff
@@ -727,6 +727,18 @@ def finalizer():
             except Exception as e:
                 print("Exception when waiting for pod deletion", e)
                 return
+            try:
```

Comment:
We have fixtures for `pv` and `pvc`, why doesn't work as expected? Want to understand the reason for failure before fixing it.

Reply:
There are some tests which are not using fixtures for pv/pvc but they are getting created in the definition of tests. 
```
test_salvage_auto_crash_all_replicas(client, core_api, volume_name, pod_make):
pod_name, pv_name, pvc_name, md5sum = \
        prepare_pod_with_data_in_mb(client, core_api, pod_make, volume_name)
```
These tests are using ```pod_make``` fixture to get pod manifest and deleting pod later. So, instead of changing tests to make them use pv/pvc fixture, a try block is added to delete pv/pvc here. This way we don't have to disturb existing tests and pv/pvc can be deleted in pod_make fixture.

Reply:
Noticed `prepare_pod_with_data_in_mb`, `create_pvc_for_volume` as well as `create_pv_for_volume` are in fact bad practise. We should always try to use the fixture first. Not sure if all the references of the above functions are really needed.

Can you file an issue to replace `create_pvc_for_volume` and `create_pv_for_volume` with `pv/pvc` fixtures in order to properly clean the objects up? We can have this PR merged first but we will still want to fix the problem from the root.

Reply:
https://github.com/longhorn/longhorn/issues/1462